File: frameworks/java/javalin/UnitTests/src/test/java/unittests/BaselineTests.java
package unittests;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertTrue;

import io.javalin.testtools.JavalinTest;
import io.javalin.testtools.Response;
import org.junit.jupiter.api.Tag;
import org.junit.jupiter.api.Test;

class BaselineTests extends JavalinApp {

    // rb:test baseline.plaintext
    @Test
    @Tag("baseline.plaintext")
    void plaintextIsTheStringAsText() {
        JavalinTest.test(app(), (server, client) -> {
            Response response = client.get("/plaintext");

            assertEquals("Hello, World!", response.body().string());
            assertTrue(Answer.header(response, "content-type").startsWith("text/plain"));
        });
    }
}

File: frameworks/dotnet/hardened/UnitTests/BaselineTests.cs
namespace UnitTests;

public class BaselineTests
{
    // rb:test baseline.plaintext
    [ModuleTest]
    [Trait("corpus", "baseline.plaintext")]
    public async Task Plaintext_is_the_text_as_plain_text(ITestWebApp app)
    {
        TestWebResponse response = await app.Get("/plaintext");

        response.Assert.Ok();
        Assert.StartsWith("text/plain", Answer.Header(response, "Content-Type"));
        Assert.Equal("Hello, World!", await response.ReadTextAsync());
    }
}
